src/components/Table/utils/TableState.ts

```diff
@@ -201,6 +211,37 @@ export class TableState {
     this.rows = rows;
   }
 
+  setColumns(columns: GridColumnWithId<any>[]): void {
+    if (columns !== this.columns) {
+      this.columns = columns;
+      this.visibleColumnStorageKey = camelCase(columns.map((c) => c.id).join());
+      this.visibleColumns.replace(readOrSetLocalVisibleColumnState(columns, this.visibleColumnStorageKey));
+      const expandedColumnIds = columns.filter((c) => c.initExpanded).map((c) => c.id);
+      this.expandedColumns.replace(expandedColumnIds);
+    }
+  }
+
+  setVisibleColumns(ids: string[]) {
+    sessionStorage.setItem(this.visibleColumnStorageKey, JSON.stringify(ids));
+    this.visibleColumns.replace(ids);
+  }
+
+  get visibleColumnIds(): string[] {
+    return [...this.visibleColumns.values()];
+  }
+
+  get expandedColumnIds(): string[] {
+    return [...this.expandedColumns.values()];
+  }
+
+  toggleExpandedColumn(columnId: string) {
```

Comment:
Question: Does toggling this require `visibleColumns` to be re-calc-d? Is that already happening somehow?

Reply:
No, it does not. `visibleColumns` only holds the top level / parent columns. Columns defined within `expandColumns` cannot be made visible/invisible right now. This was done on purpose to limit the scope of the change.

Reply:
I was trying to figure out if it made sense to have `visibleColumns` keep track of both the columns that a user toggles on and off via the "EditColumnsButton" (or similar workflow), and which columns are currently "expanded". It seemed like it could get messy with determining which is expanded + which of the potentially expanded columns are currently hidden. I was treating them as being orthogonal to one another for now. 
Maybe in a follow up we can extend this work so that the "child"/expandable columns can also be hidden via the `EditColumnsButton`. Could be cool to have that button re-render based on which columns are expanded, so you could potentially toggle on/off some of the "child" columns.

Reply:
Ah gotcha! I thought `visibleColumns` was going to be something like "if parent column A is expanded, then put its children into visibleColumns". Like more of a calculation of what columns are physically visible right now.

But what you have makes sense.

Agreed to descope on visibleColumns can only be otp-level.